dattri/algorithm/base.py

```diff
@@ -117,104 +108,127 @@ def _set_full_train_data(self, dataloader: torch.utils.data.DataLoader) -> None:
-    def generate_test_query(
+    def generate_test_rep(
         self,
-        index: int,
+        ckpt_idx: int,
         data: Tuple[torch.Tensor, ...],
     ) -> torch.Tensor:
-        """Calculating the query based on the test data.
+        """Getting the initial representations of the test data.
 
-        Inner product attributor calculates the inner product between the
-        train query and the transformation of the test query. This function
-        calculates the test query based on the test data.
+        Inner product attributor calculates the inner product between the (transformed)
+        train representations and test representations. This function generates the
+        initial test representations.
 
         Args:
-            index (int): The index of the model parameters. This index
-                is used for ensembling of different trained model
-                parameters.
-            data (Tuple[Tensor]): The test data. Normally this is a tuple
-                of input data and target data, the number of items in the
-                tuple should be aligned in the target function. The tensors'
-                shape follows (1, batchsize, ...).
+            ckpt_idx (int): The index of the model checkpoints. This index
+                is used for ensembling different trained model checkpoints.
+            data (Tuple[Tensor]): The test data. Typically, this is a tuple
+                of input data and target data but the number of items in this
+                tuple should align with the corresponding argument in the
+                target function. The tensors' shape follows (1, batch_size, ...).
 
         Returns:
-            torch.Tensor: The query based on the test data. Normally it is
-                a 2-d dimensional tensor with the shape of
-                (batchsize, num_parameters).
+            torch.Tensor: The initial representations of the test data. Typically,
+                it is a 2-d dimensional tensor with the shape of
+                (batch_size, num_parameters).
         """
-        model_params, _ = self.task.get_param(index)
+        model_params, _ = self.task.get_param(ckpt_idx)
         return self.task.get_grad_target_func()(model_params, data)
 
-    def generate_train_query(
+    def generate_train_rep(
         self,
-        index: int,
+        ckpt_idx: int,
         data: Tuple[torch.Tensor, ...],
     ) -> torch.Tensor:
-        """Calculating the query based on the train data.
+        """Generate the initial representations of the train data.
 
-        Inner product attributor calculates the inner product between the
-        train query and the transformation of the test query. This function
-        calculates the train query based on the train data.
+        Inner product attributor calculates the inner product between the (transformed)
+        train representations and test representations. This function generates the
+        initial train representations.
```

Comment:
The default implementation calculates the gradient of the train loss with respect to the parameter. Subclasses may override this function to calculate something else.